elide-core/src/main/java/com/yahoo/elide/parsers/expression/PermissionToFilterExpressionVisitor.java

```diff
@@ -78,7 +108,8 @@ public FilterExpression visitPermissionClass(ExpressionParser.PermissionClassCon
             }
             return ((FilterExpressionCheck) check).getFilterExpression(requestScope);
         } else if (UserCheck.class.isAssignableFrom(check.getClass())) {
-            return USER_CHECK_EXPRESSION;
+            boolean userCheckResult = check.ok(requestScope.getUser());
```

Comment:
I think this might be OK - BUT we have to be careful this doesn't break Xiaoyao's work to optimize user check expression evaluation.  That work uses the permission executor and has a new cache.  This would bypass that.